tokio-console/src/state/tasks.rs

```diff
@@ -217,16 +221,35 @@ impl TasksState {
                     warnings: Vec::new(),
                     location,
                 };
-                task.lint(linters);
+                if matches!(task.lint(linters), TaskLintResult::RequiresRecheck) {
+                    next_pending_lint.insert(task.id);
+                }
                 Some((id, task))
             });
 
         for (stats, mut task) in self.tasks.updated(stats_update) {
             tracing::trace!(?task, ?stats, "processing stats update for");
             task.stats = stats.into();
-            task.lint(linters);
+            if matches!(task.lint(linters), TaskLintResult::RequiresRecheck) {
+                next_pending_lint.insert(task.id);
+            } else {
+                // Avoid linting this task again this cycle
+                self.pending_lint.remove(&task.id);
+            }
         }
 
+        for id in &self.pending_lint {
+            if let Some(task) = self.tasks.get(*id) {
+                if matches!(
+                    task.borrow_mut().lint(linters),
+                    TaskLintResult::RequiresRecheck
+                ) {
+                    next_pending_lint.insert(*id);
+                }
```

Comment:
nit: It's more idiomatic Rust to use `if let` here.
```suggestion
                if let TaskLintResult::RequiresRecheck = task.borrow_mut().lint(linters) {
                    next_pending_lint.insert(*id);
                }
```

Reply:
i was going back and forth on this, thanks for the clarification!